### ISSUE TITLE
Problem with `transformResponseBody`

### ISSUE THREAD

**Issue by kebot** [h2non/rocky#94]
Here is the demo:

``` javascript
"use strict"

var PORT = 6911

var toxy = require('toxy')
, rules = toxy.rules

var proxy = toxy()

proxy
  .all('*')
  .host('api.github.com')
  .forward('https://api.github.com')
  .transformResponseBody(function (req, res, next) {
    console.log(res.body.toString())
    next(null, JSON.stringify({
      'debug': true
    }))
  })

proxy.listen(PORT)
```

then visit: `http://localhost:6911/users/facebook/repos`

After that, I can see the output in console, but nothing in the browser.

**Comment by tyler-johnson** [comment]
I have very little experience with Node streams, but I had success by calling `res.connection.uncork()` in my middleware. Seems like the connection is never being uncorked and thus never being flushed. Looking at the [Node v6 streaming source](https://github.com/nodejs/node/blob/29e49fc286080215031a81effbd59eac092fff2f/lib/_http_outgoing.js#L590) it seems like `res.end()` will only uncork if data is passed in. Rocky does not appear to pass data when it calls `res.end()`: https://github.com/h2non/rocky/blob/master/lib/middleware/response-body.js#L131.

IMO this seems like more of a Node v6 HTTP Outgoing Message issue than a Rocky issue, but a simple fix would be just to call `uncork()` before calling `end()`, or perhaps before the headers are even sent.

**Comment by SPSpwetter** [comment]
BTW, I have my local toxy built to call uncork() and it works fine now with 4.x. Commenting in the hope that will encourage h2non to address this since its a tiny change.

**Comment by h2non** [comment]
Gonna review that next week and ship a definitive fix. This issue has been there for a while, unfortunately.

**Comment by h2non** [comment]
This should be fixed in `rocky@v0.4.14`. Upgrade it and try it again.
